File: C#/第一部分 C#语言/第一章 C#语言与程序结构/第二节 C#程序结构/hello.cs
// 由bhw设计
/*
第1章 第2节 C#程序结构（Hello World 拆解）

一、程序结构核心
1) using：导入命名空间。
2) class：定义类型。
3) static void Main()：程序入口。
4) Console.WriteLine：向控制台输出文本。

二、原理说明
1) C# 代码先编译成 IL（中间语言），再由 .NET 运行时执行。
2) Main 是默认入口方法，程序从这里开始。

三、习题
1) 增加两行输出：你的名字和学习目标。
2) 把 Main 改成带参数形式：static void Main(string[] args)。

四、答案与讲解
1) 连续写 Console.WriteLine 即可。
2) args 用来接收命令行参数，后续会在参数章节讲到。
*/
using System; // 使用系统库

class Program // 定义程序类
{
    static void Main() // 程序主入口
    {
        Console.WriteLine("你好，世界！"); // 输出第一行
        Console.WriteLine("这是我的第一个 C# 程序。"); // 输出第二行
    } // Main 结束
} // 类结束
// 由bhw设计

File: C++/第一部分 C++语法/第一章 C++语言/第二节 C++程序结构/hello.cpp
// 由bhw设计
/*
第一章 第二节 C++程序结构

一、一个最小可运行C++程序通常包含
1) 头文件：提供库函数声明。
2) main函数：程序入口。
3) 语句：以分号结束。

二、程序执行顺序
1) 进入main函数。
2) 按顺序执行每一行语句。
3) 遇到return结束main并返回状态码。

三、练习题
1) 修改程序，输出你的名字和学习目标。
2) 把return 0改为return 1，思考含义有什么变化。
*/
#include <iostream> // 标准输入输出头文件
using namespace std; // 使用标准命名空间

int main() // 程序入口函数
{
    cout << "Hello, World!" << endl; // 输出经典第一行
    cout << "我叫bhw，我正在系统学习C++语法。" << endl; // 输出学习信息
    return 0; // 0表示程序正常退出
} // main函数结束

/*
练习题答案（注释版）
1) 参考输出：名字 + 目标。
2) return 1通常表示异常退出或非正常状态。

// #include <iostream> // 头文件
// using namespace std; // 命名空间
// int main() // 主函数
// {
//     cout << "我是bhw。" << endl; // 输出名字
//     cout << "目标：掌握C++基础语法。" << endl; // 输出目标
//     return 0; // 正常退出
// } // 结束
*/
// 由bhw设计
